presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java

```diff
@@ -3774,4 +3777,22 @@ public void verifyAndCleanup(SchemaTableName tableName)
             assertFalse(hdfsEnvironment.getFileSystem(context, path).exists(path));
         }
     }
+
+    @Test
+    public void testTableWithFooter()
+    {
+        try (Transaction transaction = newTransaction()) {
+            ConnectorSession session = newSession();
+            ConnectorMetadata metadata = transaction.getMetadata();
+
+            ConnectorTableHandle table = getTableHandle(metadata, tableWithFooter);
+            try {
+                metadata.getTableLayouts(session, table, new Constraint<>(TupleDomain.all(), bindings -> true), Optional.empty());
+                fail("expected exception");
+            }
+            catch (PrestoException e) {
+                assertEquals(e.getErrorCode(), HIVE_UNSUPPORTED_FORMAT.toErrorCode());
```

Comment:
check message too

also, if this check fails (eg on travis), the produced assertion error will not contain any trace of original `e`, and we will wonder.

Reply:
I added message check

Reply:
I will leave it as it is for now. This would require refactoring tests around as well.